framework/include/transfers/MultiAppGeneralFieldTransfer.h

```diff
@@ -0,0 +1,588 @@
+                               const DofobjectToInterpValVec & dofobject_to_valsvec,
+                               const InterpCaches & interp_caches);
+
+  /*
+   * Cache pointInfo
+   */
+  void cacheOutgoingPointInfo(const Point point,
+                              const dof_id_type dof_object_id,
+                              const unsigned int problem_id,
+                              ProcessorToPointVec & outgoing_points);
+
+  /**
+   * Compute minimum distance
+   * @param p the point of interest
+   * @param bbox the bounding box to find the minimum distance from
+   */
+  Real bboxMinDistance(const Point & p, const BoundingBox & bbox);
+
+  /**
+   * Compute max distance
+   * @param p the point of interest
+   * @param bbox the bounding box to find the maximum distance from
+   */
+  Real bboxMaxDistance(const Point & p, const BoundingBox & bbox);
+
+  /**
+   * @brief Obtains the max dimensions to scale all points in the mesh
+   * @return the maximum dimension in each coordinate axis of all target problems
+   */
+  Point getMaxToProblemsBBoxDimensions() const;
+
+  /**
+   * Get from bounding boxes for given domains and boundaries
+   */
+  std::vector<BoundingBox> getRestrictedFromBoundingBoxes();
+};
+
+// Anonymous namespace for data, functors to use with GenericProjector.
+namespace GeneralFieldTransfer
+{
+// Transfer::OutOfMeshValue is an actual number.  Why?  Why!
+static_assert(std::numeric_limits<Real>::has_infinity,
+              "What are you trying to use for Real?  It lacks infinity!");
+extern Number BetterOutOfMeshValue;
+
+inline bool
+isBetterOutOfMeshValue(Number val)
+{
+  // Might need to be changed for e.g. NaN
+  return val == GeneralFieldTransfer::BetterOutOfMeshValue;
+}
+
+// We need two functors that record point (value and gradient,
+// respectively) requests, so we know what queries we need to make
+// to other processors
+
+/**
+ * Value request recording base class
+ */
+template <typename Output>
+class RecordRequests
+{
+protected:
+  typedef typename TensorTools::MakeBaseNumber<Output>::type DofValueType;
+
+public:
+  typedef typename TensorTools::MakeReal<Output>::type RealType;
+  typedef DofValueType ValuePushType;
+  typedef Output FunctorValue;
+
+  RecordRequests() {}
+
+  RecordRequests(RecordRequests & primary) : _primary(&primary) {}
+
+  ~RecordRequests()
+  {
+    if (_primary)
+    {
+      Threads::spin_mutex::scoped_lock lock(Threads::spin_mtx);
+      _primary->_points_requested.insert(
+          _primary->_points_requested.end(), _points_requested.begin(), _points_requested.end());
+    }
+  }
+
+  void init_context(FEMContext &) {}
+
+  Output eval_at_node(const FEMContext &,
+                      unsigned int libmesh_dbg_var(i),
+                      unsigned int /*elem_dim*/,
+                      const Node & n,
+                      bool /*extra_hanging_dofs*/,
+                      const Real /*time*/)
+  {
+    libmesh_assert_not_equal_to(i, 0);
```

Comment:
do you remember why we have that @fdkong @roystgnr 

seems like i = 0 would be fine, would just be variable of index 0?
see https://mooseframework.inl.gov/docs/doxygen/libmesh/classlibMesh_1_1OldSolutionCoefs.html#adaffa04d151f7b42a607761d06fdb82e
for an example of this i

Reply:
`git blame` might accuse @fdkong, but that's only because he refactored code that I added.

And for the life of me I can't remember why I added it!  I don't have anything like it in any other projector code, and if anything i=0 should be the most basic case, not a disallowed case!

We ought to just delete these lines.  I'm 90% sure they're an overzealous assertion from some previous code design, and if I rediscover some circumstances under which that's a sane assertion I'll actually comment the reasoning when re-adding it and fixing whatever trips it.

Reply:
Thanks for pitching it.
I agree it's not needed. Luckily the users that should be hitting it are all in opt mode haha

Reply:
I'll make a PR soon